File: app/src/test/java/org/metabrainz/mobile/ResultItemConversionTest.java
package org.metabrainz.mobile;

import org.junit.Test;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.Artist;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.Label;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.Recording;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.Release;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.ReleaseGroup;
import org.metabrainz.mobile.presentation.features.adapters.ResultItem;
import org.metabrainz.mobile.presentation.features.adapters.ResultItemUtils;

import static org.metabrainz.mobile.AssertionUtils.checkResultItemAssertions;
import static org.metabrainz.mobile.EntityTestUtils.getTestArtist;
import static org.metabrainz.mobile.EntityTestUtils.getTestLabel;
import static org.metabrainz.mobile.EntityTestUtils.getTestRecording;
import static org.metabrainz.mobile.EntityTestUtils.getTestRelease;
import static org.metabrainz.mobile.EntityTestUtils.getTestReleaseGroup;
import static org.metabrainz.mobile.ResultItemTestUtils.getTestArtistResultItem;
import static org.metabrainz.mobile.ResultItemTestUtils.getTestLabelResultItem;
import static org.metabrainz.mobile.ResultItemTestUtils.getTestRecordingResultItem;
import static org.metabrainz.mobile.ResultItemTestUtils.getTestReleaseGroupResultItem;
import static org.metabrainz.mobile.ResultItemTestUtils.getTestReleaseResultItem;

public class ResultItemConversionTest {

    @Test
    public void testConvertArtist() {
        ResultItem testItem = getTestArtistResultItem();
        Artist artist = getTestArtist();
        ResultItem item = ResultItemUtils.getEntityAsResultItem(artist);
        checkResultItemAssertions(testItem, item);
    }

    @Test
    public void testConvertRelease() {
        ResultItem testItem = getTestReleaseResultItem();
        Release release = getTestRelease();
        ResultItem item = ResultItemUtils.getEntityAsResultItem(release);
        checkResultItemAssertions(testItem, item);
    }

    @Test
    public void testConvertLabel() {
        ResultItem testItem = getTestLabelResultItem();
        Label label = getTestLabel();
        ResultItem item = ResultItemUtils.getEntityAsResultItem(label);
        checkResultItemAssertions(testItem, item);
    }

    @Test
    public void testConvertRecording() {
        ResultItem testItem = getTestRecordingResultItem();
        Recording recording = getTestRecording();
        ResultItem item = ResultItemUtils.getEntityAsResultItem(recording);
        checkResultItemAssertions(testItem, item);
    }

    @Test
    public void testConvertReleaseGroup() {
        ResultItem testItem = getTestReleaseGroupResultItem();
        ReleaseGroup releaseGroup = getTestReleaseGroup();
        ResultItem item = ResultItemUtils.getEntityAsResultItem(releaseGroup);
        checkResultItemAssertions(testItem, item);
    }

}

File: app/src/test/java/org/metabrainz/mobile/ResultItemConversionTest.kt
package org.metabrainz.mobile

import org.metabrainz.mobile.presentation.features.adapters.ResultItemUtils.getEntityAsResultItem
import org.junit.Test
import org.metabrainz.mobile.EntityTestUtils.testArtist
import org.metabrainz.mobile.EntityTestUtils.testLabel
import org.metabrainz.mobile.EntityTestUtils.testRecording
import org.metabrainz.mobile.EntityTestUtils.testRelease
import org.metabrainz.mobile.EntityTestUtils.testReleaseGroup

class ResultItemConversionTest {
    @Test
    fun testConvertArtist() {
        val testItem = ResultItemTestUtils.testArtistResultItem
        val item = getEntityAsResultItem(testArtist)
        AssertionUtils.checkResultItemAssertions(testItem, item)
    }

    @Test
    fun testConvertRelease() {
        val testItem = ResultItemTestUtils.testReleaseResultItem
        val item = getEntityAsResultItem(testRelease)
        AssertionUtils.checkResultItemAssertions(testItem, item)
    }

    @Test
    fun testConvertLabel() {
        val testItem = ResultItemTestUtils.testLabelResultItem
        val item = getEntityAsResultItem(testLabel)
        AssertionUtils.checkResultItemAssertions(testItem, item)
    }

    @Test
    fun testConvertRecording() {
        val testItem = ResultItemTestUtils.testRecordingResultItem
        val item = getEntityAsResultItem(testRecording)
        AssertionUtils.checkResultItemAssertions(testItem, item)
    }

    @Test
    fun testConvertReleaseGroup() {
        val testItem = ResultItemTestUtils.testReleaseGroupResultItem
        val item = getEntityAsResultItem(testReleaseGroup)
        AssertionUtils.checkResultItemAssertions(testItem, item)
    }
}